includes/models/class-wc-stripe-subscription.php

```diff
@@ -0,0 +1,298 @@
+	const META_STRIPE_CARD_ID = '_stripe_card_id';
+
+	/**
+	 * Meta key for the delayed update payment method.
+	 *
+	 * @var string
+	 */
+	const META_DELAYED_UPDATE_PAYMENT_METHOD_ALL = '_delayed_update_payment_method_all';
+
+	/**
+	 * Meta key for the Stripe refund ID.
+	 *
+	 * @var string
+	 */
+	const META_STRIPE_REFUND_ID = '_stripe_refund_id';
+
+	/**
+	 * The subscription object.
+	 *
+	 * @var WC_Subscription
+	 */
+	private $wc_subscription;
+
+	/**
+	 * Constructor.
+	 *
+	 * @param $subscription WC_Subscription|WC_Stripe_Subscription The subscription object.
+	 */
+	public function __construct( $subscription ) {
+		if ( $subscription instanceof WC_Stripe_Subscription ) {
+			$this->wc_subscription = $subscription->wc_subscription;
+		} elseif ( class_exists( 'WC_Subscription' ) && $subscription instanceof WC_Subscription ) {
+			$this->wc_subscription = $subscription;
+		}
+
+		if ( ! $this->wc_subscription ) {
+			$type = is_object( $subscription ) ? get_class( $subscription ) : gettype( $subscription );
+			throw new InvalidArgumentException( 'WC_Stripe_Subscription requires a valid underlying WC_Subscription; supplied $subscription argument was ' . $type );
+		}
+	}
+
+	/**
+	 * Magic method to call methods on the underlying WC_Subscription object.
+	 *
+	 * @param string $name The name of the method to call.
+	 * @param array $arguments The arguments to pass to the method.
+	 */
+	public function __call( $name, $arguments ) {
+		if ( method_exists( 'WC_Subscription', $name ) ) {
+			return call_user_func_array( [ $this->wc_subscription, $name ], $arguments );
+		}
+
+		throw new BadMethodCallException( 'Method ' . $name . ' does not exist in WC_Stripe_Subscription' );
+	}
+
+	/**
+	 * Wrapper to return an order using the extension's custom WC_Stripe_Subscription class.
+	 *
+	 * @param $subscription_id int Subscription ID.
+	 * @return bool|WC_Stripe_Subscription
+	 */
+	public static function get_by_id( $subscription_id ) {
+		$subscription = function_exists( 'wcs_get_subscription' ) ? wcs_get_subscription( $subscription_id ) : false;
+		return $subscription ? self::to_instance( $subscription ) : false;
+	}
+
+	/**
+	 * Wrapper to get subscriptions using the extension's custom WC_Stripe_Subscription class.
+	 *
+	 * @param $args array Arguments to pass to wcs_get_subscriptions.
+	 * @return array|WC_Stripe_Subscription[]
+	 */
+	public static function query( $args ) {
+		$subscriptions = function_exists( 'wcs_get_subscriptions' ) ? wcs_get_subscriptions( $args ) : [];
+		if ( empty( $subscriptions ) ) {
+			return [];
+		}
+
+		return array_map(
+			function ( $subscription ) {
+				return self::to_instance( $subscription );
+			},
+			$subscriptions
+		);
+	}
+
+	/**
+	 * Wrapper to get subscriptions for an order using the extension's custom WC_Stripe_Subscription class.
+	 *
+	 * @param $order int|WC_Order Order object or ID.
+	 * @return array|WC_Stripe_Subscription[]
+	 */
+	public static function get_for_order( $order ) {
```

Comment:
Naming suggestion - this more closely mirrors the underlying `wcs_get_subscriptions_for_order()`.
```suggestion
	public static function get_subscriptions_for_order( $order ) {
```

Reply:
See https://github.com/woocommerce/woocommerce-gateway-stripe/pull/4254/#discussion_r2071631006